### PR TITLE
chore: consolidate config files

### PR DESCRIPTION
**Pull request by longlho**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,9 +8,11 @@
     "url": "https://github.com/formatjs/formatjs"
   },
   "scripts": {
+    "clean": "rimraf packages/*/dist packages/*/lib",
     "build:bootstrap": "lerna bootstrap --hoist --since --include-filtered-dependencies --no-ci",
-    "build:packages": "lerna run build --since --stream",
-    "build": "npm run build:bootstrap && npm run build:packages",
+    "build:packages": "npm run clean && lerna run clean && lerna run build --since --stream",
```
Comment (redonkulus) [inline package.json]: I think lerna clean asks for confirmation to delete. Add `—yes` to avoid the prompt.

File: mocha.opts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+-r ts-node/register
+-r source-map-support/register
+-r chai/register-expect
+--recursive
+packages/**/tests/index.ts
```
Comment (redonkulus) [inline mocha.opts]: We don’t really want to test every package every build. We should just leverage lerna and only run for what changed.

File: package.json [hunk 3]
```diff
@@ -19,7 +21,8 @@
     "karma:local": "karma start karma.conf.js",
     "lint": "eslint .",
     "prettier": "prettier --write 'packages/*/{lib,test,tests,src}/**/*.ts' '{src,scripts,test,tests}/*.js'",
-    "test": "lerna run lint --since --stream && lerna run test --since --stream"
+    "test": "lerna run lint --since --stream && npm run mocha && lerna run test --since --stream",
```
Comment (redonkulus) [inline package.json]: It runs mocha and lerna run test?

File: packages/intl-format-cache/package.json [hunk 4]
```diff
@@ -7,11 +7,12 @@
...
     "benchmark": "ts-node --project tests/tsconfig.json tests/benchmark",
-    "build": "tsc -p src && tsc -p src/tsconfig.cjs.json",
-    "clean": "rimraf lib dist",
-    "cover": "nyc npm t",
-    "prepublishOnly": "npm run clean && npm run build",
-    "test": "TS_NODE_PROJECT='tests/tsconfig.json' mocha -r ts-node/register -r source-map-support/register -r chai/register-expect tests/index.ts --reporter spec"
+    "build": "tsc && tsc -p src/tsconfig.cjs.json",
+    "prepublishOnly": "npm run build"
+  },
+  "devDependencies": {
```
Comment (redonkulus) [inline packages/intl-format-cache/package.json:15]: Remove

File: packages/intl-format-cache/.gitignore [hunk 5]
```diff
@@ -1,4 +1,10 @@
 lib/
 node_modules/
 .nyc_output
-dist
\ No newline at end of file
+dist
+src/*.js
```
Comment (redonkulus) [inline packages/intl-format-cache/.gitignore]: Ignore source?

**Review by redonkulus — CHANGES_REQUESTED** [review]
I think the downside to rollup and mocha consolidation is that you basically run the build and test of all packages regardless of whether the packages changed. This could lead to unnecessary patch publishes by lerna. 

For rollup and mocha, you can just update the run script of each package to point to the shared mocha.opts and rollup config at the monorepo root.

I like leveraging lerna to only build what changed and its dependencies instead of all the things.

File: packages/intl-format-cache/package.json (hunk 4 above)
Comment (longlho) [reply packages/intl-format-cache/package.json:15]: This is actually needed so that lerna runs these packages before `intl-format-cache`

**Review by redonkulus — APPROVED** [review]

**Pull request merged**